backends/bmv2/common/deparser.cpp

```diff
@@ -24,14 +24,53 @@ void DeparserConverter::convertDeparserBody(const IR::Vector<IR::StatOrDecl>* bo
                                           Util::JsonArray* order, Util::JsonArray* primitives) {
     ctxt->conv->simpleExpressionsOnly = true;
     for (auto s : *body) {
+        auto isR = false;
+        IR::MethodCallExpression *mce2 = nullptr;
         if (auto block = s->to<IR::BlockStatement>()) {
             convertDeparserBody(&block->components, order, primitives);
             continue;
         } else if (s->is<IR::ReturnStatement>() || s->is<IR::ExitStatement>()) {
             break;
         } else if (s->is<IR::EmptyStatement>()) {
             continue;
-        } else if (s->is<IR::MethodCallStatement>()) {
+        } else if (s->is<IR::AssignmentStatement>()) {
```

Comment:
Should these changes be in common/deparser.cpp? They are specific to PSA, and not common with v1model for example. Can you find a way to move these changes to the PSA backend without duplicating too much code?

Reply:
I put this in common for several reasons:

1. If someone wants to use `InternetChecksum` in simple switch and v1model, program will fail before it gets here, because ss arch doesn't recognize it.
2. To avoid redundant code.
3. Other structures that are only for one architecture are also used in common section.. for example:
https://github.com/p4lang/p4c/blob/2c2c928e972f92dd5fb1de9509e7bd4d91c354ca/backends/bmv2/common/action.cpp#L63-L80
https://github.com/p4lang/p4c/blob/2c2c928e972f92dd5fb1de9509e7bd4d91c354ca/backends/bmv2/common/parser.cpp#L206-L214

Reply:
Perhaps, but this is actually bad for long term maintenance. People who will modify this code in the future will be confused about why this is here, as I am now. The fact that it was done before does not mean it's right, it means that the review didn't catch it.
If you really insist having this code here please add a commend explaining that it is in fact intended just for the PSA back-end.

Reply:
In fact I was also confused when I saw structures that are not common in common folder. I will add this part of code in `psaSwitch.cpp.`

Reply:
I added comments. 
In current `p4c PSA switch` implementation this is the best approach to preserve performance and avoid the same processing multiple times (processing parsers - first time for common structures and second only for `InternetChecksum`).